### DISCUSSION TITLE
confusion: first/second pass vs single/double passs

### DISCUSSION THREAD

**Discussion by JM-Bader** [vdemichev/DiaNN#411] [Q&A] [answered]
Frankly, I am a bit confused about i) first/second pass (steps of the search)  vs ii) single/double pass (neural network classifier)

i): 
![image](https://user-images.githubusercontent.com/98888664/174584706-849bd5a9-c23f-4689-8629-fe089e7533ff.png)

ii): 
![image](https://user-images.githubusercontent.com/98888664/174584559-9aa50e2c-63cd-498b-bf29-7069b48a5805.png)

 vs ii) single/double pass (neural network classifier)

Q1: Is there a relationship between the i) and ii)? 
(First I thought that it was actually the same thing but then set ii) to single-pass according to your recommendation in #399. However, I still got first-pass and second-pass (doesn't say second pass in screenshot above) output files)

Q2: ii) can be set in the DIANN GUI. i) apparantly can't (can it?). This contributes to the confusion.
Q3: Regarding ii),  your recommendation in #399  is to use the single-pass neural network classifier (at least still in DIANN1.8.1) as the double-pass is experimental, correct?
Q4: Regarding i), what is your recommendation? Should one use the first pass results or the second pass results?
Q5: What do i) and ii) actually do? 
Regarding i) Based on the log of searches it appears that a the results of an initial search (first pass) are used to refine the library and then re-search the raw files resulting in the second pass results. Correct?
Regarding ii) - it is not clear what step of DIANN this pertains to. Could you clarify ? 
Where do I find more information about this?

**Comment by vdemichev** [comment, marked as answer]
Q1: no.
Q2: 'first-pass' output files correspond to the first pass of MBR.
Q3: Yes.
Q4: Always use the main output, 'first-pass' results are an extra bit.
Q5: First: yes, correct. NN double pass means peak group selection using NNs.
